source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -68,11 +70,16 @@ def getCommentInfoFromPosition(position):
 		UIAElement=UIAElementArray.getElement(index)
 		UIAElement=UIAElement.buildUpdatedCache(UIAHandler.handler.baseCacheRequest)
 		obj=UIA(UIAElement=UIAElement)
-		if not obj.parent or obj.parent.name!='Comment':
+		if (obj.parent.role !=controlTypes.ROLE_GROUPING and not
+		obj.parent.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_IsAnnotationPatternAvailablePropertyId)):
 			continue
 		comment=obj.makeTextInfo(textInfos.POSITION_ALL).text
 		dateObj=obj.previous
 		date=dateObj.name
+		if not dateObj.previous:

```

Comment:
I think that it is more readable if you check obj.previous.previous first. Now, you are assigning an object to the dateObj variable and then, when it doesn't seem to be a dateObj, you're assigning the same object to authorObj.

For example:

```
		comment=obj.makeTextInfo(textInfos.POSITION_ALL).text
		tempObj = obj.previous.previous
		authorObj = tempObj else obj.previous
		author =authorObj.name
		if not tempObj:
			return dict(comment=comment,author=author)
		dateObj=obj.previous
		date=dateObj.name
		return dict(comment=comment,author=author,date=date)
```